### PR TITLE
Mypy 141 plus fixes

### PR DESCRIPTION
**Pull request by glyph**
upgrade to mypy 1.4.1 and update a few annotations

### CODE REVIEW

File: src/klein/test/test_form.py [hunk 1]
```diff
@@ -72,6 +73,13 @@ def procureASession(self, request: IRequest) -> Any:
     def danglingParameter(self, dangling: str) -> None:
         "..."
 
+    @requirer.require(
+        router.route("/dangling-param", methods=["GET"]),
+        form=Form.rendererFor(danglingParameter, action="/dangling-param"),
+    )
+    def renderDanglingParameter(self, form: Form) -> Form:
+        return form
```
Comment (adiroiban) [inline src/klein/test/test_form.py:90]: I think that a short docstring for `def danglingParameter(self, dangling: str) -> None:` on line 73 as well as here, would help.

But maybe for people familiar with klein, there is no need to explain what a `dangling` parameter is.

I tried to search the docs for `dangling`... no search results

https://klein.readthedocs.io/en/latest/search.html?q=dangling&check_keywords=yes&area=default

File: src/klein/test/test_form.py [hunk 2]
```diff
@@ -365,7 +373,22 @@ def test_noName(self) -> None:
         errors = self.flushLoggedErrors(ValueError)
         self.assertEqual(len(errors), 1)
         self.assertIn(
-            str(errors[0].value), "Cannot extract unnamed form field."
+            "Cannot extract unnamed form field.",
+            str(errors[0].value),
+        )
+
+        self.successResultOf(
```
Comment (adiroiban) [inline src/klein/test/test_form.py:392]: the docstring for `test_noName` is quite brief...
I am not sure if this new code is in the same scope, or if this requires a separate test method, with a separate docstring.

**Review by adiroiban — APPROVED** [review]
Changes look good. Thanks.

I don't understand the new tests...  but the existing tests don't look good either :) ... so nothing is getting worse here.

I guess that it can be merged.

cheers

File: src/klein/test/test_form.py (hunk 1 above)
Comment (glyph) [reply src/klein/test/test_form.py:90]: Dangling fields are a really weird edge case which we need to test for thoroughness (they require code to handle them) but it's really just a kind of mistake that users *could* make, but rarely would have the opportunity to.  I'll add more `@see L{DanglingField}` to all the relevant places so users can see what all the issues are.

File: src/klein/test/test_form.py (hunk 2 above)
Comment (glyph) [reply src/klein/test/test_form.py:392]: (Same issue as above, so handling the same way).
